Polygon_mesh_processing/include/CGAL/Polygon_mesh_processing/internal/Corefinement/intersection_nodes.h

```diff
@@ -284,7 +285,15 @@ class Intersection_nodes<TriangleMesh,VertexPointMap,true,false>
   {
     put(vpm, vd, exact_to_double(enodes[i]));
     if (&tm1==&tm)
-      tm1_vertices[i] = vd;
+    {
+      if (  tm1_vertices[i] == GT::null_vertex() )
+      {
+        tm1_vertices[i] = vd;
+        return;
+      }
+      if (doing_autorefinement)
+        tm2_vertices[i] = vd;
+    }
```

Comment:
I have replaced `==` by `=`, but please @sloriot have a look at the code, to check that the line should modify `tm2_vertices` and not `tm1_vertices`.

Reply:
Thanks for the fix, it is the correct one (tm2 needs to be updated).